### PR TITLE
REALMC-6875: Add native Object fromEntries & values methods

### PR DESCRIPTION
**Pull request by makesitgo**

### CODE REVIEW

File: otto_test.go [hunk 1]
```diff
@@ -838,7 +839,22 @@ func TestAPI(t *testing.T) {
...
+		expectedValues := []interface{}{1, false, math.NaN()}
+		is(object.Keys(), expectedKeys)
+
+		values := object.Values()
+		is(len(values), len(expectedValues))
+		for i, value := range values {
+			is(value, expectedValues[i])
+		}
+
+		// this assert fails:
```
Comment (makesitgo) [inline otto_test.go:853]: ⚠️ ... any ideas to get this working?

Comment (tsedgwick) [reply otto_test.go:853]: it's probably doing a pointer check and seeing as they aren't matching. Do we gain much by doing a whole slice check vs the iterative check above on the values?

File: builtin_object.go [hunk 2]
```diff
@@ -275,6 +275,55 @@ func builtinObject_keys(call FunctionCall) Value {
...
+		})
+
+		return toValue_object(call.runtime.newArrayOf(entries))
+	}
+	panic(call.runtime.panicTypeError())
+}
+
+func builtinObject_fromEntries(call FunctionCall) Value {
+	res := call.runtime.newObject()
+	if object := call.Argument(0)._object(); nil != object {
```
Comment (tsedgwick) [inline builtin_object.go:308]: [nit] flip this conditional to remove a nested block

File: builtin_object.go [hunk 3]
```diff
@@ -275,6 +275,55 @@ func builtinObject_keys(call FunctionCall) Value {
...
+			values = append(values, object.get(name))
+			return true
+		})
+		return toValue_object(call.runtime.newArrayOf(values))
+	}
+	panic(call.runtime.panicTypeError())
+}
+
+func builtinObject_entries(call FunctionCall) Value {
+	if object := call.Argument(0)._object(); nil != object {
```
Comment (tsedgwick) [inline builtin_object.go:290]: [nit] flip this conditional to remove a nested block

File: builtin_object.go [hunk 4]
```diff
@@ -275,6 +275,55 @@ func builtinObject_keys(call FunctionCall) Value {
 	panic(call.runtime.panicTypeError())
 }
 
+func builtinObject_values(call FunctionCall) Value {
+	if object, values := call.Argument(0)._object(), []Value(nil); nil != object {
```
Comment (tsedgwick) [inline builtin_object.go:279]: [nit] flip this conditional to remove a nested block

**Review by tsedgwick — APPROVED** [review]
lgtm outside a few nits

File: builtin_object.go (hunk 2 above)
Comment (makesitgo) [reply builtin_object.go:308]: while I don't disagree with the suggestion, this is a fairly consistent pattern throughout this entire file.  I'm going to elect to keep them as is here for sake of preserving the code patterns...

**Pull request merged**